### PR TITLE
Upgrade Calico to 3.25.0

### PR DESCRIPTION
**Pull request by neoaggelos**
### Summary

Upgrade Calico to 3.25.0

### Changes

- Get new calico manifests from 3.25.0 release
- Update docs for how to customize calico manifest
- Remove the separate s390x flavour, upstream Calico supports s390x out of the box now, see manifest below.

### Testing

- Single node
- Cluster with vxlan.
- Cluster with BGP.

<details><summary> calico/node:v3.25.0 architectures </summary>

```
$ docker manifest inspect calico/node:v3.25.0
{
   "schemaVersion": 2,
   "mediaType": "application/vnd.docker.distribution.manifest.list.v2+json",
   "manifests": [
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 2214,
         "digest": "sha256:0745f9a52bef94dc9f80d78e705726d95e03e9f04ced65226af63da422c39656",
         "platform": {
            "architecture": "arm",
            "os": "linux",
            "variant": "v7"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 737,
         "digest": "sha256:56db28c3632192f56a1ff1360b83ef640fc8f41fa21a83126194811713e2f022",
         "platform": {
            "architecture": "amd64",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 2425,
         "digest": "sha256:5edeb4c9f9c3e8a1d1c7c50d174ca6511b902710a901636e94767762bacbb012",
         "platform": {
            "architecture": "s390x",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 944,
         "digest": "sha256:ced1834e93fd9d7ab7752fd4cf7d645d77f3b0940cb3e7acf71549d44a62d04b",
         "platform": {
            "architecture": "arm64",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 2424,
         "digest": "sha256:538b47778eea82d6d37c11002dabd290e004843bdd8117bbd6b2fc3769882fa3",
         "platform": {
            "architecture": "ppc64le",
            "os": "linux"
         }
      }
   ]
}
```

</details>

### CODE REVIEW

**Review by berkayoz — APPROVED** [review]
LGTM!

File: upgrade-scripts/000-switch-to-calico/resources/calico.yaml [hunk 1]
```diff
@@ -4548,7 +4684,7 @@ spec:
             path: /var/snap/microk8s/current/var/run/calico
         - name: var-lib-calico
           hostPath:
-            path: /var/snap/microk8s/current/var/lib/calico
+            path: /var/snap/microk8s/common/var/lib/calico
```
Comment (ktsakalozos) [inline upgrade-scripts/000-switch-to-calico/resources/calico.yaml:4687]: Why do we switch from current to common?

**Comment by ktsakalozos** [comment]
I am afraid we may need to update the [calico/upgrade.py](https://github.com/canonical/microk8s/blob/master/scripts/calico/upgrade.py) script to force the same cni.yaml across all nodes of the same version.

When the snap refreshes we [reload the cni](https://github.com/canonical/microk8s/blob/master/snap/hooks/configure#L663). In case of a cluster where some of its nodes got updated to a new version (eg they went from 1.26 to 1.27) and some others that are fresh installs (ie snap installed to 1.27) we would have a mixture of old and new calico cnis and old ones. During a refresh, depending on which node gets refreshed the cluster will bounce between the two versions.

**Comment by neoaggelos** [comment]
I would say that we are fine. We do not apply the CNI again, we rather just rollout restart the pods https://github.com/canonical/microk8s/blob/34412f66e0ca456215163b26adeef3f65251cf13/microk8s-resources/default-hooks/reconcile.d/10-cni-restart#L15-L25

**Pull request merged**